### PR TITLE
[WebAssembly] Build more of musl with libc.py

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

**Comment by sbc100** [comment]
The enables the following change on the waterfall side: https://github.com/WebAssembly/waterfall/compare/cleanup_musl_build?expand=1

**Comment by sbc100** [comment]
Actaully please hold off on landing this until I've verified there no regressions.

File: arch/wasm32/wasm.js
Comment (jfbastien) [inline arch/wasm32/wasm.js:831]: Seems like you delete things, aren't they needed anymore?

File: arch/wasm32/wasm.js
Comment (jfbastien) [inline arch/wasm32/wasm.js:846]: I expect longjmp to get lowered. Same for setjmp later. We shouldn't see them here?

File: arch/wasm32/wasm.js
Comment (jfbastien) [inline arch/wasm32/wasm.js:879]: Surprising to see sem and pthread things here, but I guess they're not implemented for threading yet? You probably don't want them in `musl_hack`.

File: arch/wasm32/wasm.js
Comment (jfbastien) [inline arch/wasm32/wasm.js]: This seems incorrect?

File: libc.py
Comment (jfbastien) [inline libc.py]: Nice!

File: arch/wasm32/wasm.js
Comment (sbc100) [reply arch/wasm32/wasm.js:831]: This whole section is auto-generated based on the above command... so its precisely what is needed by musl.wast

File: arch/wasm32/wasm.js
Comment (sbc100) [reply arch/wasm32/wasm.js]: Hmm.. since __syscall is declared as varargs.  But it does need to exist.  It ends up being used for syscalles with 7 args since there is no __syscall7.   See ./src/internal/syscall.h.  For an example of a callsite see ./src/fcntl/posix_fadvise.c which ends up being a 7 arg call.

File: arch/wasm32/wasm.js
Comment (sbc100) [reply arch/wasm32/wasm.js:879]: As I said, this is auto-generated.      I could add the pthread sources too, which would reduce this.. but not sure if we want to be including musl's pthread impl.

**Comment by sbc100** [comment]
I think this is good to go now.

File: arch/wasm32/pthread_arch.h
Comment (jfbastien) [inline arch/wasm32/pthread_arch.h]: In a wasm embedding this will ~never cause a crash. Could you instead return the largest valid pointer, so it's guaranteed to crash if used?

**Review by jfbastien — APPROVED** [review]
One thing then lgtm.

File: arch/wasm32/pthread_arch.h
Comment (binji) [reply arch/wasm32/pthread_arch.h]: huh, cool idea! Kinda makes me wonder how broken everything would be if we always treat `nullptr` as `-1`...

Comment (jfbastien) [reply arch/wasm32/pthread_arch.h]: Totally valid to do! 😁

**Comment by sbc100** [comment]
I think this is mergable now?    Do you want to give me commit access to this repo so I can merge my own changes once approved?

**Comment by jfbastien** [comment]
@sbc100 sent you an invite.

**Pull request merged**
